draft-west-cookie-incrementalism.md

```diff
@@ -401,85 +404,361 @@ from "have the same registrable domain" (or "is an exact match for") to say
+
+We should remedy this defect by storing both a "scheme" and "port"
+component along with the cookie, and use those components in cookies'
+matching algorithms to ensure that a cookie is only sent to the origin
+that originally set it, thus keeping origins' state separate by
+default.
+
+Example: 
+
+* `https://example.com` sets a cookie `foo=https` and `http://example.com`
+   sets a cookie `foo=http`. Previously this would result in one of
+   the cookies being overwritten by the other. Now, they're considered
+   seperate and when visiting each site users could see their
+   respective cookie. The same holds true for `https://example.com` and
+   `https://example.com:444`, each is a different origin and thus have
+   their own cookies.
+
+It's possible that a site operator purposefully wants some cookies to
+be accessible across port boundaries. Because this would weaken the
+origin boundary, by increasing the cookie's scope, we'd want this to be
+an opt-in mechanism. Helpfully there is already an attribute that
+increases a cookie's scope: the `Domain` attribute. We can modify the
+`Domain` attribute slightly such that it not only allows cookies to be
+accessible by different hosts but also by different ports than the
+origin's that set it. For convenience we'll call any cookie with a
+`Domain` attribute a "domain cookie".
+
+Example:
+
+* A corporate network, `https://corp.example`, has various services
+   each running on their own port. Those services share use of a token
+   which allows users to use the services. To get a token a user logs
+   in on `https://corp.example/login/` which creates the cookie
+   `IDToken=a1b2bc3`. Next, to file an expense, the user needs to
+   visit `https://corp.example:8443`, but when they do the IDToken
+   isn't sent because the ports differ (443 vs 8443) and the user is
+   denied access. To remedy this, the IDToken can have the `Domain` 
+   attribute added to it: `IDToken=a1b2d3; Domain=corp.example`.
+   Now when the user visits `https://corp.example:8443` the token
+   is sent and access is granted as expected.
+
+Because any domain cookie is now exposed to multiple origins it means
+a cookie created by one origin can be overwritten by another origin. 
+
+Examples:
+
+1. `https://example.com:123` sets a cookie: `Set-Cookie: 
+    foo=domaincookie; Domain=example.com`. The user then visits 
+    `https://example.com:456` which sets the same cookie. This second
+    cookie would match the first and then overwrite it. 
+2. `https://example.com:123` sets a cookie 
+    `Set-Cookie: foo=domaincookie; Domain=example.com`. The user then
+    visits `https://example.com:456` which sets `Set-Cookie: 
+    foo=origincookie`. The second cookie would match the first an
+    overwrite it.
+    
+Also, because domain cookies are less trusted due to their wider scope,
+we'll want to avoid allowing them to shadow non-domain cookies by
+disallowing domain cookies from co-exiting or overwriting a matching
```

Comment:
Nit: "co-exiting" seems like a typo (or a way of describing multiple cookies sent in the same outgoing request that I haven't heard before?) Perhaps "co-existing with"?